docs/apps/build-system.adoc

```diff
@@ -89,33 +89,42 @@ NOTE:  XP apps deployed via file (locally) are presented with a small blue icon
 
 === Development Mode
 
-Enonic XP supports a so-called Development mode.
-When running in this mode, XP will automatically source JavaScript controllers directly from your project folder.
+[WARNING]
+====
+As the name implies, development mode is intended only for development use and should _not_ be used in production. Doing so would cause both security and performance issues.
+====
 
-This may be convenient if you are developing with pure JavaScript and do not depend on build steps in your project.
+Enonic XP can be run in a special mode known as _development mode_. In this mode, XP will detect any changes to your project source files, including JavaScript controllers and XML schemas. It does this by reading these files directly from their source locations instead of from the application `.jar` file.
 
-You may activate dev mode when starting XP as follows:
+Development mode also disables some of XP's caching mechanisms. (What mechanisms and why?)
```

Comment:
- assetUrl function generates new fingerprint for every request, so browsers can't cache assets
- resource cache invalidated if resource `modified time`changes.
- entire JavaScript `require` cache invalidated  if any of required resources  `modified time`changes.